File: src/main/java/dev/psygamer/wireframe/entity/LivingEntity.java
package dev.psygamer.wireframe.entity;

import dev.psygamer.wireframe.internal.entity.InternalLivingEntity;

public interface LivingEntity extends Entity {
	
	static LivingEntity get(final net.minecraft.entity.LivingEntity internalEntity) {
		if (internalEntity == null)
			return null;
		
		return new InternalLivingEntity(internalEntity);
	}
	
	/* Health */
	
	int getMaxHealthLevel();
	
	int getHealthLevel();
	
	void setHealthLevel(final int healthLevel);
	
	/* Leashing */
	
	boolean canBeLeashedTo(Player player);
	
	boolean isLeashedTo(Player player);
	
	Player getLeashHolder();
	
	void setLeashHolder(Player player);
	
	void unleash();
	
	/* Internal */
	
	@Override
	net.minecraft.entity.LivingEntity getInternal();
}

File: src/main/kotlin/dev/psygamer/wireframe/entity/LivingEntity.kt
package dev.psygamer.wireframe.entity

interface LivingEntity : Entity {
	
	val maxHealth: Int
	var health: Int
	
	var leashHolder: Player?
	
	fun canBeLeashedTo(player: Player): Boolean
	fun isLeashedTo(player: Player): Boolean
	
	fun unleash() {
		unleash(true)
	}
	
	fun unleash(dropLeashItem: Boolean)
}